packages/vscode-extension/src/registerDisposables.ts

```diff
@@ -0,0 +1,146 @@
+import { Neo4jDataSummary, PollingEvent } from '@neo4j-cypher/schema-poller';
+import {
+  commands,
+  ConfigurationChangeEvent,
+  Disposable,
+  Uri,
+  window,
+  workspace,
+} from 'vscode';
+import { getLanguageClient } from './appContext';
+import {
+  Connection,
+  deleteConnection,
+  getConnection,
+  getCurrentConnection,
+  saveConnection,
+  testConnection,
+  toggleConnection,
+  updateConnectionDataSummary,
+} from './connection';
+import { ConnectionPanel } from './connectionPanel';
+import {
+  ConnectionItem,
+  ConnectionTreeDataProvider,
+} from './connectionTreeDataProvider';
+import * as constants from './constants';
+import {
+  MethodName,
+  sendNotificationToLanguageClient,
+} from './languageClientService';
+
+export function registerDisposables(extensionUri: Uri): Disposable[] {
+  const disposables = Array<Disposable>();
+  const connectionTreeDataProvider = new ConnectionTreeDataProvider();
+  const client = getLanguageClient();
+
+  disposables.push(
+    window.registerTreeDataProvider(
+      'neo4jConnections',
+      connectionTreeDataProvider,
+    ),
+    commands.registerCommand(
+      constants.TEST_CONNECTION_COMMAND,
+      async (connection: Connection, password: string) => {
+        if (await testConnection(connection, password)) {
+          void window.showInformationMessage(
+            constants.TEST_CONNECTION_SUCCESFUL_MESSAGE,
+          );
+        } else {
+          void window.showErrorMessage(constants.CONNECTION_FAILED_MESSAGE);
+        }
+      },
+    ),
+    commands.registerCommand(
+      constants.SAVE_CONNECTION_COMMAND,
+      async (
+        connection: Connection,
+        password: string,
+        isNew: boolean,
+      ): Promise<boolean> => {
+        if (await testConnection(connection, password)) {
```

Comment:
something to consider.. do we want to avoid testing the connection each time we add it 

maybe we want to keep track of whether a connection has been tested before saving though this might make the form more complex.. it will reduce overhead on the driver